VerilogCodingStyle.md

```diff
@@ -797,6 +797,36 @@ endmodule : foo
 localparam bit ValBaz = 1;  // This comment describes the item to the left.
 ```
 
+It can sometimes be useful to structure the code using header-style comments in
+order to separate different functional parts (like FSMs, the main datapath or
+registers) within a module. In that case, the preferred style is as follows:
+
+```systemverilog
+module foo;
+
+  ////////////////
+  // Controller //
+  ////////////////
+  ...
+
+  ///////////////////////
+  // Main ALU Datapath //
+  ///////////////////////
+  ...
+
+endmodule : foo
+```
+
+If the designer would like to use comments to mark the beginning/end of a
+particular section for better readability (e.g. in nested for loop blocks), the
+preferred way is to use a single-line comment:
+
+```
+// begin: section
+...
+// end: section
```

Comment:
I'm not sure I fully understand that one. Is "section" meant to be replaced by something, or the whole thing, or is section a keyword?

Reply:
yes this can be replaced freely, should I change to `<my section id>` or similar ?

Reply:
so people are expected to use `begin: my_fsm` or `begin: a awesome code block`? (so an ID, or a description?) Maybe just use an example instead of "section".

Reply:
they are not expected to do that on every block, only if they feel that it helps to read the code.

in case they want to add such comments, however, they should use single-line comments similar to the ones shown, without other fancy delineators inside the comment.

let me rephrase this.